src/libstd/sys/unix/fs.rs

```diff
@@ -459,9 +459,18 @@ pub fn rmdir(p: &Path) -> io::Result<()> {
 }
 
 pub fn readlink(p: &Path) -> io::Result<PathBuf> {
+    #[cfg(not(target_libc = "newlib"))]
```

Comment:
Hm, where'd this new `target_libc` attribute come from? This overlaps with the current `target_env` argument and we may not want to stabilize on this just yet.